### ISSUE TITLE
Playground for this configuration

### ISSUE THREAD

**Issue by anshulguleria** [yannickcr/eslint-plugin-react#1589]
Is there any playground for this plugin so that I can verify if the issue is really there?

I was having an issue where I was using a variable in className which was not defined but I got no syntax error. I wanted to confirm this on playground first before reporting it. So that's why I am asking if there is any place where I can confirm first?

Now regarding my issue:

I am using [material-ui](https://material-ui-next.com/) library for building my UI. And [classnames]() for adding multiple classes to react

```javascript
class MyClassComponent extends Component {
    render () {
        const { classes } = this.props;
        return (
            <AppBar className={
                    classnames({ [classes.shiftedAppBar]: open })
                }>
                Some code here
            </AppBar>
        );
    }
}
const styles = () => {
    shiftedAppBar: {}
}
export default withStyles(styles)(MyClassComponent);
``` 

Here in above code `open` should raise an error because its not defined anywhere but it's not.

I am using rules as:
```
        "react/jsx-uses-react": "error",
        "react/jsx-uses-vars": "error",
        "react/react-in-jsx-scope": "error",
        "react/jsx-no-undef": "error"
```
__Update:__

It does show error as expected if the name of the variable is something other than `open` but why not `open` too?

**Comment by ljharb** [comment]
In this case, `open` is used in a JS context, so eslint-plugin-react wouldn't pay attention to it. Have you enabled the core `no-undef` rule?

**Comment by anshulguleria** [comment]
Hi @ljharb  I am using 
```
extends: "eslnt:recommended"
```
That has `no-undef` applied by default.

**Comment by ljharb** [comment]
hopefully `eslint` not `eslnt`.

Ah - what if you name the not-defined variable something other than "open", which is a valid global variable in browsers?

**Comment by anshulguleria** [comment]
Ooh, I understood it. So `open` is a valid global variable that's why it was not raising an error. The same thing happens if I use `location` or any other valid global variable.

Is there any way I can make sure that global variables are used with their contexts like `window.open` or `global.open`?

That can make things more declarative.

**Comment by silvenon** [comment]
The problem there is that there are lots of globals like `Math` that you probably want to use implicitly, so ESLint can't really know where to draw the line. You could use [no-restricted-globals](https://eslint.org/docs/rules/no-restricted-globals) to catch specific mistakes that you often make (I use it to prohibit accidentally using native `Map` and `Set` instead their Immutable.js counterparts).

**Comment by anshulguleria** [comment]
Thanks. That could help, although not completely but better than nothing. 😄

**Comment by ljharb** [comment]
Sounds like the motivating issue is closed; as far as a playground, I think it'd be better for eslint itself to offer a playground that allows including plugins from npm rather than each plugin trying to solve this problem separately.

**Comment by silvenon** [comment]
@anshulguleria if you have an idea for a new rule, you can suggest it in the ESLint repo, maybe I could also help. Then we can continue the discussion there.

**Comment by silvenon** [comment]
Sorry for posting here again, just one more thing. @anshulguleria know I believe what you're looking for is [eslint-restricted-globals](https://github.com/doshisid/eslint-restricted-globals), a sensible collection of globals to forbid.

**Comment by anshulguleria** [comment]
@silvenon  Thanks that could really help.
